### PR TITLE
[CU-20185x7] Add integration test mosaic implementation.

### PR DESCRIPTION
**Pull request by Juliansv**

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-20185x7 Mosaic Pallet Integration Tests](https://app.clickup.com/t/20185x7)

**Review by Dom-Roth — CHANGES_REQUESTED** [review]
~~There is a type definition missing for `PalletMosaicDecayLinearDecayer`, please add it to `runtime-tests/src/types/interfaces/crowdloanRewards/definitions.ts`. Else the tests won't run.
You probably already added it but didn't commit.~~

NVM i was on an old version!

File: integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+            transferAmount = 100000000000;
+            networkId = 1;
+            pNetworkId = api.createType('u128', 1);
+            ethAddress = "0x";
+            remoteAssetId = api.createType('CommonMosaicRemoteAssetId', {
+                EthereumTokenAddress: api.createType('[u8;20]', "0x")
+            });
+        });
+        before('Mint available assets into wallets', async function(){
+            this.timeout(2*60*1000);
```
Comment (Dom-Roth) [inline integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts:72]:
```suggestion
            this.timeout(5*60*1000);
```
Please increase the timeout a bit. It seems to occasionally fail therefore.

File: integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+         * Sudo call therefore result is checked by `.isOk`.
+         */
+        it('Should be able to rotate relayer', async function () {
+            if (!testConfiguration.enabledTests.query.account__success.balanceGTZero1)
+               this.skip();
+            const {data:[result]} = await TxMosaicTests.testRotateRelayer(startRelayerWallet, 
+                newRelayerWallet.address);
+            const relayerInfo = await api.query.mosaic.relayer();
+            //verify that the relayer records information about the next relayer wallet
+            expect(relayerInfo.unwrap().relayer.next.toJSON().account).to.be.equal(api.createType('AccountId32', newRelayerWallet.address).toString());
```
Comment (Dom-Roth) [inline integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts:200]: This only compiles if the code is compiled without type checking cause the linter seems to block this. 
I couldn't get it quickly fixed. We can try on this together tomorrow.

```
> tsc -p . --declaration

test/tests/mosaic/mosaicTests.ts:200:41 - error TS2339: Property 'relayer' does not exist on type 'PalletMosaicRelayerStaleRelayer'.

200             expect(relayerInfo.unwrap().relayer.next.toJSON().account).to.be.equal(api.createType('AccountId32', newRelayerWallet.address).toString());
                                            ~~~~~~~


Found 1 error in test/tests/mosaic/mosaicTests.ts:200
```

File: integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+import { send } from 'process';
+import {
+    CommonMosaicRemoteAssetId,
+    OrmlTokensAccountData,
+    PalletMosaicDecayBudgetPenaltyDecayer,
+    PalletMosaicNetworkInfo
+} from "@composable/types/interfaces";
+import {start} from "repl";
+import {AccountId} from "@polkadot/types/interfaces/runtime";
+import { doesNotMatch } from 'assert';
```
Comment (Dom-Roth) [inline integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts:16]: A lot of unused imports, can be reduced to:
```suggestion
import { expect } from 'chai';
import testConfiguration from './test_configuration.json';
import { sendAndWaitForSuccess } from '@composable/utils/polkadotjs';
import {KeyringPair} from "@polkadot/keyring/types";
import { mintAssetsToWallet } from '@composable/utils/mintingHelper';
import {
    CommonMosaicRemoteAssetId,
    OrmlTokensAccountData,
    PalletMosaicDecayBudgetPenaltyDecayer,
    PalletMosaicNetworkInfo
} from "@composable/types/interfaces";
```

**Review by Dom-Roth — CHANGES_REQUESTED** [review]
Looks really good so far! A few tweaks, as well as that weird compile issue, and we're ready to go.

**Review by Dom-Roth — APPROVED** [review]

**Review by kollegian — APPROVED** [review]
LGTM

**Review by Dom-Roth — CHANGES_REQUESTED** [review]
Commits need to be signed!

**Review by Dom-Roth — APPROVED** [review]

**Review by kollegian — APPROVED** [review]
LGTM

**Pull request merged**